hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt

```diff
@@ -650,8 +652,205 @@ class AttributeSearchIntegrationTest : AbstractSearchDataIntegrationTest() {
+      )
+
+      val request = RequestDsl {
+        query {
+          booleanMatcher("recall" IS false)
+        }
+      }
+
+      webTestClient.attributeSearch(request).expectPrisoners("B1234BB")
+    }
+
+    @Test
+    fun `booleans with AND`() {
+      loadPrisoners(
+        PrisonerBuilder(prisonerNumber = "A1234AA", recall = true, profileInformation = youthOffender(true)),
+        PrisonerBuilder(prisonerNumber = "B1234BB", recall = true, profileInformation = youthOffender(false)),
+        PrisonerBuilder(prisonerNumber = "C1234CC", recall = false, profileInformation = youthOffender(true)),
+        PrisonerBuilder(prisonerNumber = "D1234DD", recall = false, profileInformation = youthOffender(false)),
+      )
+
+      val request = RequestDsl {
+        query {
+          booleanMatcher("recall" IS true)
+          booleanMatcher("youthOffender" IS false)
+        }
+      }
+
+      webTestClient.attributeSearch(request).expectPrisoners("B1234BB")
+    }
+
+    @Test
+    fun `booleans with OR`() {
+      loadPrisoners(
+        PrisonerBuilder(prisonerNumber = "A1234AA", recall = true, profileInformation = youthOffender(true)),
+        PrisonerBuilder(prisonerNumber = "B1234BB", recall = true, profileInformation = youthOffender(false)),
+        PrisonerBuilder(prisonerNumber = "C1234CC", recall = false, profileInformation = youthOffender(true)),
+        PrisonerBuilder(prisonerNumber = "D1234DD", recall = false, profileInformation = youthOffender(false)),
+      )
+
+      val request = RequestDsl {
+        query {
+          joinType = OR
+          booleanMatcher("recall" IS true)
+          booleanMatcher("youthOffender" IS true)
+        }
+      }
+
+      webTestClient.attributeSearch(request).expectPrisoners("A1234AA", "B1234BB", "C1234CC")
+    }
+
+    @Test
+    fun `booleans AND strings AND integers`() {
+      loadPrisoners(
+        PrisonerBuilder(prisonerNumber = "A1234AA", recall = true, firstName = "John", heightCentimetres = 170),
+        PrisonerBuilder(prisonerNumber = "B1234BB", recall = true, firstName = "Jack", heightCentimetres = 170),
+        PrisonerBuilder(prisonerNumber = "C1234CC", recall = true, firstName = "John", heightCentimetres = 171),
+        PrisonerBuilder(prisonerNumber = "D1234DD", recall = false, firstName = "John", heightCentimetres = 170),
+        PrisonerBuilder(prisonerNumber = "E1234EE", recall = false, firstName = "Jack", heightCentimetres = 170),
+        PrisonerBuilder(prisonerNumber = "F1234FF", recall = false, firstName = "John", heightCentimetres = 171),
+      )
+
+      val request = RequestDsl {
+        query {
+          booleanMatcher("recall" IS false)
+          stringMatcher("firstName" IS "John")
+          intMatcher("heightCentimetres" EQ 171)
+        }
+      }
+
+      webTestClient.attributeSearch(request).expectPrisoners("F1234FF")
+    }
+
+    @Test
+    fun `booleans OR strings OR integers`() {
```

Comment:
Just Date and DateTime types to support. But we'll have to get selective with the combinations to test from now on because 5 matcher types each with AND/OR, multiple queries and subQueries means we can't test every combination ever.